src/sentry/static/sentry/app/views/alerts/rules/details/index.tsx

```diff
@@ -1,160 +1,77 @@
 import React from 'react';
 import {RouteComponentProps} from 'react-router';
 
-import {markIncidentAsSeen} from 'app/actionCreators/incident';
-import {addErrorMessage} from 'app/actionCreators/indicator';
 import {fetchOrgMembers} from 'app/actionCreators/members';
 import {Client} from 'app/api';
 import Feature from 'app/components/acl/feature';
-import {t} from 'app/locale';
 import {Organization} from 'app/types';
-import {trackAnalyticsEvent} from 'app/utils/analytics';
 import withApi from 'app/utils/withApi';
+import {IncidentRule} from 'app/views/settings/incidentRules/types';
 
-import {Incident, IncidentStats, IncidentStatus} from '../../types';
-import {
-  fetchIncident,
-  fetchIncidentStats,
-  isOpen,
-  updateStatus,
-  updateSubscription,
-} from '../../utils';
+import {Incident} from '../../types';
+import {fetchAlertRule, fetchIncidentsForRule} from '../../utils';
 
 import DetailsBody from './body';
 import DetailsHeader from './header';
 
 type Props = {
   api: Client;
   organization: Organization;
-} & RouteComponentProps<{alertId: string; orgId: string}, {}>;
+} & RouteComponentProps<{ruleId: string; orgId: string}, {}>;
 
 type State = {
   isLoading: boolean;
   hasError: boolean;
-  incident?: Incident;
-  stats?: IncidentStats;
+  rule?: IncidentRule;
+  incidents?: Incident[];
 };
 
 class AlertRuleDetails extends React.Component<Props, State> {
   state: State = {isLoading: false, hasError: false};
 
   componentDidMount() {
-    const {api, organization, params} = this.props;
-
-    trackAnalyticsEvent({
-      eventKey: 'alert_details.viewed',
-      eventName: 'Alert Details: Viewed',
-      organization_id: parseInt(organization.id, 10),
-      alert_id: parseInt(params.alertId, 10),
-    });
+    const {api, params} = this.props;
 
     fetchOrgMembers(api, params.orgId);
-
     this.fetchData();
   }
 
   fetchData = async () => {
     this.setState({isLoading: true, hasError: false});
-
     const {
       api,
-      params: {orgId, alertId},
+      params: {orgId, ruleId},
     } = this.props;
 
     try {
-      const incidentPromise = fetchIncident(api, orgId, alertId).then(incident => {
-        this.setState({incident});
-        markIncidentAsSeen(api, orgId, incident);
+      const rulePromise = fetchAlertRule(api, orgId, ruleId).then(rule => {
```

Comment:
nit: don't need the `{` `}` here because it's a single line function

Reply:
👍